### DISCUSSION TITLE
Are cross links possible ?

### DISCUSSION THREAD

**Discussion by 573** [srid/neuron#50] [Ideas]
Will it be possible to reference other zettelkastens using the z: protocol ? I. e. using
```
[2011406](z://neuron.srid.ca)
```
instead of [Creating and Editing zettels 2011406](https://neuron.srid.ca/2011406.html).

**Comment by srid** [comment]
In order to visually distinguish (in a more obvious manner) regular z: links from cross links, the later could probably look like this:

```markdown
[neuron.srid.ca/2011406](z:)
```

And rendered in HTML using "neuron.srid.ca/2011406" (in place of the bare ID) as the link text.

---

But I fail to see the value of adding this feature. What's the real-world use case for linking between Zettelkastens using z: links (instead of normal links)?

**Comment by 573** [comment]
Nevermind, just thought about having a convenient way to refer to other peoples' zettelkastens.
Maybe that is out-of-scope of the zettelkasten idea anyways.

**Comment by srid** [comment]
I'm opening this, because I find myself wanting this feature lately. :-)

If I add a regular link to an external zettelkasten like this:

```markdown
Users are delighted of it: <https://haskell.zettel.page/2016101.html>
```

It currently renders like this:

![image](https://user-images.githubusercontent.com/3998/79877459-40a60800-83ba-11ea-97ce-d161d6d07594.png)

But it would be great if it rendered like:

![image](https://user-images.githubusercontent.com/3998/79877618-71863d00-83ba-11ea-87b0-5bd522abe035.png)

**Comment by dpwiz** [comment]
I would suggest [InterWiki](https://en.wikipedia.org/wiki/Interwiki_links)-style links, where each server is responsible for its map.

This would allow for [shorter](https://en.wikipedia.org/wiki/Petname) and more stable cross-linking.
I.e. somebody migrates their ZK to a new domain and everybody else is left with either broken links or search-and-replace task on their hands.

In addition, this would allow some server-to-server functionality later. Maybe even [Fediverse](https://en.wikipedia.org/wiki/Fediverse) integration.

**Comment by dpwiz** [comment]
And the maps don't have to be `http`! One can even uplift their instance to IPFS/GNUnet namespaces, while keeping notes intact and content-addressable for everybody.

**Comment by srid** [comment]
@dpwiz Interesting ideas. Today I can imagine doing the following:

1. Add an "interMap" key to `neuron.dhall` where you specify a mapping of `$siteName` to `$siteBaseUrl`. Like InterWiki configuration.

2. In the Markdown, support a special linking syntax `<haskell:2016101>` which links to the 2016101 zettel on the external zettelkasten aliased with siteName=haskell and siteBaseUrl=haskell.zettel.page.

---

Interestingly, the interMap syntax `<haskell:...>` is consistent with the `<z:zettels...>` of https://github.com/srid/neuron/issues/59#issuecomment-616158997

**Reply by srid** [reply]
[Neuron NG](https://note.ema.srid.ca/) has an interesting feature based on the union filesystem of Linux. Where you can "merge" multiple Zettelkastens to present an unified view (and it hot-reloads when any of the zk is updated). 

This doedsn't address cross links per se, but links like `[[Foo/Bar]]` are supported as well, to help disambiguate things. Special cross links syntax might be one solution but since they wouldn't have editor support they may not be ideal.
